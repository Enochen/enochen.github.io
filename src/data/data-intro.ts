import { IconType } from 'react-icons'
import {
  FaFlask,
  FaGithub,
  FaGraduationCap,
  FaLinkedin,
  FaNewspaper,
  FaRegUser,
  FaUser,
} from 'react-icons/fa'

export interface IIcon {
  icon: IconType
  iconAlt?: IconType
  label: string
  link?: string
  external?: boolean
}

export interface IIntro {
  name: string
  desc?: string
  aboutData?: string[]
  iconData?: IIcon[]
}

export const IntroData: IIntro = {
  name: 'Enoch Chen',
  desc: 'cs @ cornell',
  aboutData: [
    'I am currently an undergrad at Cornell University studying Computer Science. I am passionate about designing effective and reliable systems. My goal is to make things that others can enjoy well.',
    'In my free time, I like tinkering with keyboards.',
    'Check out some of my stuff <a href = "#projects">here</a>.',
  ],
  iconData: [
    {
      icon: FaUser,
      iconAlt: FaRegUser,
      label: 'About',
    },
    {
      icon: FaFlask,
      label: 'Projects',
      link: '#projects',
    },
    {
      icon: FaGithub,
      label: 'Github',
      link: 'http://github.com/Enochen',
      external: true,
    },
    {
      icon: FaLinkedin,
      label: 'LinkedIn',
      link: 'https://www.linkedin.com/in/enochen728/',
      external: true,
    },
    {
      icon: FaGraduationCap,
      label: 'Transcript',
      link: '/misc/2020-01-04-Transcript.pdf',
      external: true,
    },
    {
      icon: FaNewspaper,
      label: 'Resume',
      link: '/misc/Enoch Chen resume.pdf',
      external: true,
    },
  ],
}
