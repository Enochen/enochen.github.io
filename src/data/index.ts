export * from './data-intro'
export * from './data-project'