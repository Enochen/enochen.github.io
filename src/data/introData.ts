import { IconType } from 'react-icons'
import {
  FaFlask,
  FaGithub,
  FaGraduationCap,
  FaLinkedin,
  FaNewspaper,
  FaRegUser,
  FaUser,
} from 'react-icons/fa'

interface IIcon {
  icon: IconType
  icon2?: IconType
  label: string
  link?: string
  external?: boolean
}

interface IIntro {
  name: string
  desc: string
  aboutData: string[]
  iconData: IIcon[]
}

export const IntroData: IIntro = {
  name: 'Enoch Chen',
  desc: 'cs @ cornell',
  aboutData: [
    'I am currently an undergrad at Cornell University studying Computer Science. I am passionate about designing effective and reliable systems. My goal is to make things that others can enjoy well.',
    'In my free time, I like tinkering with keyboards.',
    'Check out some of my stuff <a href = "#projects">here</a>.',
  ],
  iconData: [
    {
      icon: FaUser,
      icon2: FaRegUser,
      label: 'About',
    },
    {
      icon: FaFlask,
      label: 'Projects',
      link: '#projects',
    },
    {
      icon: FaGithub,
      label: 'Github',
      link: 'http://github.com/Enochen',
      external: true,
    },
    {
      icon: FaLinkedin,
      label: 'LinkedIn',
      link: 'https://www.linkedin.com/in/enochen728/',
      external: true,
    },
    {
      icon: FaGraduationCap,
      label: 'Transcript',
      link: '/misc/2020-01-04-Transcript.pdf',
      external: true,
    },
    {
      icon: FaNewspaper,
      label: 'Resume',
      link: '/misc/Enoch Chen resume.pdf',
      external: true,
    },
  ],
}
