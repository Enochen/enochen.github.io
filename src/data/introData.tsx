import React from 'react'
import { OutboundLink } from 'gatsby-plugin-gtag'
import { IconType } from 'react-icons'

import {
  FaFlask,
  FaGithub,
  FaGraduationCap,
  FaLinkedin,
  FaNewspaper,
  FaRegUser,
  FaUser,
} from 'react-icons/fa'

export interface IconInfo {
  icon: IconType
  iconAlt?: IconType
  label: string
  link?: string
  external?: boolean
}

export interface IntroInfo {
  name: string
  desc?: string
  aboutData?: JSX.Element[]
  iconData?: IconInfo[]
}

export const IntroData: IntroInfo = {
  name: 'Enoch Chen',
  desc: 'cs @ cornell',
  aboutData: [
    <>
      I am currently an undergrad at Cornell University studying Computer
      Science. I am passionate about designing effective and reliable systems,
      and I love getting the details just right. I want to make things that
      others can enjoy!
    </>,
    <>
      In my free time, you can find me juggling, tinkering with keyboards, or
      just thinking about life.
    </>,
    <>
      Check out some of my stuff{' '}
      <OutboundLink href="#projects">here</OutboundLink>.
    </>,
  ],
  iconData: [
    {
      icon: FaRegUser,
      iconAlt: FaUser,
      label: 'About',
    },
    {
      icon: FaFlask,
      label: 'Projects',
      link: '#projects',
    },
    {
      icon: FaGithub,
      label: 'Github',
      link: 'http://github.com/Enochen',
      external: true,
    },
    {
      icon: FaLinkedin,
      label: 'LinkedIn',
      link: 'https://www.linkedin.com/in/enochen728/',
      external: true,
    },
    {
      icon: FaNewspaper,
      label: 'Resume',
      link: '/misc/Enoch Chen resume.pdf',
      external: true,
    },
  ],
}
