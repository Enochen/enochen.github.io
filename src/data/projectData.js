export default [
  {
    name: "cue",
    desc:
      "I joined <a href='https://cornelldti.org' target='_blank' rel='noopener'>Cornell Design & Tech Initiative</a> in Fall 2019 as a Full Stack Developer. My team is responsible for cue, an event-discovery app for Cornell students at Ithaca. I developed a set of scripts to facilitate populating our event database with external data.",
    tags: ["HTML/CSS", "Javascript", "Python", "ReactJS", "Django"],
    live: "https://www.cornelldti.org/Projects/events",
    github: "https://github.com/cornell-dti/events-backend/",
    image: "cue.jpg",
  },
  {
    name: "Ask Palette",
    desc:
      "Generate color palettes from your search terms! Uses Google Cloud Vision to extract colors from search terms. Developed as a hackathon project at <a href='https://bigredhacks.com' target='_blank' rel='noopener'>BigRed//Hacks</a>.",
    tags: ["HTML/CSS", "Python", "Flask", "Google Cloud"],
    live: "https://askpalette.appspot.com/",
    github: "https://github.com/Enochen/askpalette",
    image: "askpalette.png",
  },
  {
    name: "Half Bot Half Brain",
    desc:
      "It's like Mad Libs, but better. The backend generates grammatically correct text from input with the help of spaCy. Developed as a hackathon project at <a href='https://teenhacksli.com/' target='_blank' rel='noopener'>TeenhacksLI</a>.",
    tags: ["HTML/CSS", "JavaScript", "Python", "spaCy"],
    live: "https://halfbothalfbrain.netlify.com/",
    github: "https://github.com/ageor0114/nyit-hackathon",
    image: "hbhb.png",
  },
]