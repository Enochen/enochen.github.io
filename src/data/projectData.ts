export interface ProjectInfo {
  name: string
  desc: string
  tags: string[]
  live?: string
  github?: string
  image: string
}

export interface ProjectList {
  data: ProjectInfo[]
}

export const ProjectData: ProjectList = {
  data: [
    {
      name: 'cue',
      desc:
        'I joined <a href="https://cornelldti.org" target="_blank" rel="noopener">Cornell Design & Tech Initiative</a> in Fall 2019 as a Full Stack Developer. My team is responsible for cue, an event-discovery app for Cornell students at Ithaca. I developed a set of scripts to facilitate populating our event database with external data.',
      tags: ['HTML/CSS', 'Javascript', 'Python', 'ReactJS', 'Django'],
      live: 'https://www.cornelldti.org/Projects/events',
      github: 'https://github.com/cornell-dti/events-backend/',
      image: 'cue.jpg',
    },
    {
      name: 'Ask Palette',
      desc:
        'Generate color palettes from your search terms! Uses Google Cloud Vision to extract colors from search terms. Developed as a hackathon project at <a href="https://bigredhacks.com" target="_blank" rel="noopener">BigRed//Hacks</a>.',
      tags: ['HTML/CSS', 'Python', 'Flask', 'Google Cloud'],
      live: 'https://askpalette.appspot.com/',
      github: 'https://github.com/Enochen/askpalette',
      image: 'askpalette.png',
    },
    {
      name: 'This Website',
      desc:
        'Over winter break, I decided to learn some new technologies and use them to put something together. It seemed like a good idea to make a personal website, and this is what came out of that idea.',
      tags: ['HTML/CSS', 'TypeScript', 'GatsbyJS', 'GraphQL', 'React/Preact' ],
      github: 'https://github.com/Enochen/portfolio',
      image: 'portfolio.png',
    },
    {
      name: 'Half Bot Half Brain',
      desc:
        'It\'s like Mad Libs, but better. The backend generates grammatically correct text from input with the help of spaCy. Developed as a hackathon project at <a href="https://teenhacksli.com/" target="_blank" rel="noopener">TeenhacksLI</a>.',
      tags: ['HTML/CSS', 'JavaScript', 'Python', 'spaCy'],
      live: 'https://halfbothalfbrain.netlify.com/',
      github: 'https://github.com/ageor0114/nyit-hackathon',
      image: 'hbhb.png',
    },
  ],
}
