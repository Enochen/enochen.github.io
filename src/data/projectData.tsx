import React from 'react'
import { OutboundLink } from 'gatsby-plugin-gtag'

export interface ProjectInfo {
  name: string
  desc: JSX.Element
  tags: string[]
  live?: string
  github?: string
  image: string
}

export interface ProjectList {
  data: ProjectInfo[]
}

export const ProjectData: ProjectList = {
  data: [
    {
      name: 'Moody',
      desc: (
        <>
          Mood-tracking app with a chatbot that sends you memes! What began as
          an offhand comment became a wacky twist on the mental health tech
          space. Hackathon Finalist at{' '}
          <OutboundLink
            href="https://hackourcampus.com/"
            target="_blank"
            rel="noopener"
          >
            HackOurCampus
          </OutboundLink>
          .
        </>
      ),
      tags: ['TypeScript', 'React Native', 'Google Cloud DialogFlow'],
      github: 'https://github.com/Enochen/hoc20-mobile',
      image: 'moody.jpg',
    },
    {
      name: '#outbreak;;',
      desc: (
        <>
          Pick a disease, pick a starting country, and buy upgrades to infect the whole
          world before a vaccine is found! Plague Inc-inspired game with a React
          web frontend and a backend server written in OCaml. Developed by a
          team of 3 for the final project of CS 3110 (Functional Programming).
        </>
      ),
      tags: ['OCaml', 'React', 'HTML/CSS', 'JavaScript'],
      github: 'https://github.com/ptwu/outbreak',
      image: 'outbreak.png',
    },
    {
      name: 'eve',
      desc: (
        <>
          I joined{' '}
          <OutboundLink
            href="https://cornelldti.org"
            target="_blank"
            rel="noopener"
          >
            Cornell Design & Tech Initiative
          </OutboundLink>{' '}
          in Fall 2019 as a Full Stack Developer. My team is responsible for
          eve, an event-discovery app for Cornell students at Ithaca. I've had a
          hand in many parts of the project, from helpful external scripts to an
          authentication system.
        </>
      ),
      tags: ['TypeScript', 'Express', 'Firebase', 'React'],
      live: 'https://www.cornelldti.org/Projects/events',
      github: 'https://github.com/cornell-dti/events-backend/',
      image: 'eve.jpg',
    },
    {
      name: 'Ask Palette',
      desc: (
        <>
          Generate color palettes from your search terms! Uses Google Cloud
          Vision to extract colors from search terms. Developed as a hackathon
          project at{' '}
          <OutboundLink
            href="https://bigredhacks.com"
            target="_blank"
            rel="noopener"
          >
            BigRed//Hacks
          </OutboundLink>
          .
        </>
      ),
      tags: ['HTML/CSS', 'Python', 'Flask', 'Google Cloud Vision'],
      live: 'https://askpalette.appspot.com/',
      github: 'https://github.com/Enochen/askpalette',
      image: 'askpalette.png',
    },
    {
      name: 'This Website',
      desc: (
        <>
          Over winter break, I decided to learn some new technologies and use
          them to put something together. It seemed like a good idea to make a
          personal website, and this is what came out of that idea.
        </>
      ),
      tags: ['HTML/CSS', 'TypeScript', 'GatsbyJS', 'GraphQL', 'React'],
      github: 'https://github.com/Enochen/portfolio',
      image: 'portfolio.png',
    },
    {
      name: 'Half Bot Half Brain',
      desc: (
        <>
          It\'s like Mad Libs, but better. The backend generates grammatically
          correct text from input with the help of spaCy. Developed as a
          hackathon project at{' '}
          <OutboundLink
            href="https://teenhacksli.com/"
            target="_blank"
            rel="noopener"
          >
            TeenhacksLI
          </OutboundLink>
          .
        </>
      ),
      tags: ['HTML/CSS', 'JavaScript', 'Python', 'spaCy'],
      live: 'https://halfbothalfbrain.netlify.com/',
      github: 'https://github.com/ageor0114/nyit-hackathon',
      image: 'hbhb.png',
    },
  ],
}
