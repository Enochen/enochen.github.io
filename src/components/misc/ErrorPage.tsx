import React, { FC } from 'react'
import { IntroInfo } from '../../data/introData'
import * as styled from '../Intro/Intro.styled'

export const ErrorPage: FC<IntroInfo> = ({ name, desc }) => {
  return (
    <styled.Intro>
      <styled.Header>
        <styled.BaseNameGroup>
          <styled.BaseName>{name}</styled.BaseName>
          <styled.BaseDesc>{desc}</styled.BaseDesc>
        </styled.BaseNameGroup>
      </styled.Header>
      <styled.IconGroup></styled.IconGroup>
    </styled.Intro>
  )
}
