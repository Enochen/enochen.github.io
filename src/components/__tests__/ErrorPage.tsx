import React from 'react'
import renderer from 'react-test-renderer'
import { IntroInfo } from '../../data'
import { ErrorPage } from '../misc/ErrorPage'
describe('Error Page', () => {
  it('renders correctly', () => {
    const errorData: IntroInfo = {
      name: 'Whoopsiess',
      desc: '40404 we did a whoooopsiess',
    }
    const tree = renderer.create(<ErrorPage {...errorData} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
