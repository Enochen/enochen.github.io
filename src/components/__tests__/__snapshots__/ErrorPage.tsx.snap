// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Page renders correctly 1`] = `
<div
  className="sc-AykKC cyoTwy sc-fzXfLU dYFJPF"
>
  <div
    className="sc-AykKF bUlJgU"
  >
    <div
      className="sc-AykKG eLNwia"
    >
      <h1
        className="sc-AykKI jZVfIj"
      >
        Whoopsiess
      </h1>
      <h2
        className="sc-AykKK eFuClL"
      >
        40404 we did a whoooopsiess
      </h2>
    </div>
  </div>
  <div
    className="sc-fzXfLQ fkALgY"
  />
</div>
`;
